runtime/tests/Test_LightningMeasures.cpp

```diff
@@ -79,7 +79,7 @@ TEMPLATE_LIST_TEST_CASE("Measurement collapse test with 2 wires", "[Measures]",
     std::vector<QubitIdType> Qs = sim->AllocateQubits(n);
 
     sim->NamedOperation("Hadamard", {}, {Qs[0]}, false);
-    auto m = sim->Measure(Qs[0]);
+    auto m = sim->Measure(Qs[0], -1);
```

Comment:
@maliasadi @rauletorresc This sort of code is unfortunate and what I would consider "smelly", having to specify `-1` values whenever the `measure` function is used, hence my earlier suggestion. I think it's smelly for two reasons:
- having to specify a value at all:
  
  If I simply want to measure without optional features I shouldn't have to specify a value for a flag I don't care about. With more flags and more optional features it's easy to see why this is suboptimal.

- having to supply a value of -1:
  
  Without context, `-1` is a meaningless value. It may be a necessary evil for C code due to the age of the language, but since this is C++ we have access to types that can more meaningfully distinguish between the presence of a value and the value itself.
  
Having said that, it's nothing I would hold up the PR for but something to consider in the future.

Reply:
You are right. I'll write an issue about this.

Reply:
@dime10 the point is rather obvious and we should always avoid writing C-like code in our C++ codebase. I want to say that we unfortunately missed it in our reviews among other points and discussions! Thanks for caching this though :)  I'll provide a quick fix for it.

Reply:
Thanks, Ali